### PR TITLE
Core Algorithms delicate-cow

### PR DESCRIPTION
**Pull request by TrevorJamesH**

### CODE REVIEW

File: src/collatz.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import dataTypeCheck from '../src/dataTypeCheck'
+
+export default function collatz(number) {
+
+  if(!dataTypeCheck(number,'number')) {
+    return "Wrong Data Type"
+  }
+
+  let collatzArray = []
```
Comment (pllearns) [inline src/collatz.js:9]: Great job so far. Look at new ways of refactoring, even if it's a google search or two.

File: src/isPalindrome.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import dataTypeCheck from '../src/dataTypeCheck'
+
+export default function isPalindrome(input) {
+
+  if(!dataTypeCheck(input,'string')) {
+    return "Wrong Data Type"
+  }
+
+  let firstString = input.toUpperCase().split('').filter((n)=>n.match(/[A-Za-z0-9]/)).join('')
+  let secondString = firstString.split('').reverse().join('')
```
Comment (pllearns) [inline src/isPalindrome.js:10]: Great work implementing key js methods. Awesome.

File: test/bubbleSortRecursion_test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      expect(bubbleSort([456,9027,98])).to.eql([98,456,9027])
+    })
+  })
+
+  context("when given [1,2,3,4,5,6,7,8,9,10,9,8,7,6,5,4,3,2,1]", function(){
+    it("should return [1,1,2,2,3,3,4,4,5,5,6,6,7,7,8,8,9,9,10]", function(){
+      expect(bubbleSort([1,2,3,4,5,6,7,8,9,10,9,8,7,6,5,4,3,2,1])).to.eql([1,1,2,2,3,3,4,4,5,5,6,6,7,7,8,8,9,9,10])
+    })
+  })
+})
```
Comment (pllearns) [inline test/bubbleSortRecursion_test.js:33]: Good job in fleshing out test cases.

File: test/bubbleSort_test.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  it('should be a function', function() {
+    expect(bubbleSort).to.be.a('function')
+  })
+
+  it('should sort through indices of an array', function() {
+    expect(bubbleSort([4,3,2,1])).to.eql([1,2,3,4])
+    expect(bubbleSort([10,2,7,5,8,3,6,1,4,9])).to.eql([1,2,3,4,5,6,7,8,9,10])
+    expect(bubbleSort([10])).to.eql([10])
+  })
+})
```
Comment (pllearns) [inline test/bubbleSort_test.js:14]: Try to add another edge case, and one that returns an error.

**Review by pllearns — COMMENTED** [review]
Great work Trevor! Always be testing and find ways to implement additional edge cases. Let me know if you got all of the tests passing. Great work!!!

File: src/dataTypeCheck.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  if(expectedType === 'object' && typeof data === 'object' && !Array.isArray(data)) {
+    return true
+  }
+  if(expectedType === 'string' && typeof data === 'string') {
+    return true
+  }
+  if(expectedType === 'number' && !isNaN(data)) {
+    return true
+  }
+  return false
```
Comment (Anasauce) [inline src/dataTypeCheck.js:14]: I would try to refactor this function and reduce the amount of "if" statements. Define concisely what value you are looking for and if that value type does not match then return "wrong value type". 

You could possibly get rid of it all together by just defining the specific datatype you are looking for and all other responses that do not match with return "wrong datatype" 

``` const dataTypeCheck = //acceptable array```

File: src/connectedGraph.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+export default function connectedGraph(graph){
+
+  if(!dataTypeCheck(graph,'object')){
+    return "Wrong Data Type"
+  }
+  let allPoints = Object.keys(graph)
+  let connected = []
+  let toCheck = [allPoints[0]]
+
+  while(toCheck.length>0) {
```
Comment (Anasauce) [inline src/connectedGraph.js:13]: just a stylistic note...keeping spacing consistent throughout your program is good practice. and in general space between operators is easier to read and can reduce bugs :)

File: src/isPalindrome.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import dataTypeCheck from '../src/dataTypeCheck'
+
+export default function isPalindrome(input) {
+
+  if(!dataTypeCheck(input,'string')) {
+    return "Wrong Data Type"
+  }
+
+  let firstString = input.toUpperCase().split('').filter((n)=>n.match(/[A-Za-z0-9]/)).join('')
```
Comment (Anasauce) [inline src/isPalindrome.js:9]: spacing between operators :)

File: src/setCompliment.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+export default function setCompliment(array, compliment) {
+
+  if(!dataTypeCheck(array,'array')) {
+    return "Wrong Data Type"
+  }
+
+let result = []
+  for(let i=0;i<compliment.length;i++) {
+    let toAdd = true
+    for(let j=0;j<array.length;j++) {
```
Comment (Anasauce) [inline src/setCompliment.js:11]: I'm not seeing a need to define a temp variable here for ```toAdd```

maybe:

```
for( let j=0; j < array.length; j++ ) {
  if( compliment[i] !=== array[j] ) {
    result.push( compliment[i] )
  }
}
 return result
```

File: src/setCompliment.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import dataTypeCheck from '../src/dataTypeCheck'
+export default function setCompliment(array, compliment) {
+
+  if(!dataTypeCheck(array,'array')) {
+    return "Wrong Data Type"
+  }
+
+let result = []
+  for(let i=0;i<compliment.length;i++) {
```
Comment (Anasauce) [inline src/setCompliment.js:9]: spaces between operators, easier readability

File: src/setIntersection.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import dataTypeCheck from '../src/dataTypeCheck'
+export default function setInstersection(array, intersect) {
+
+  if(!dataTypeCheck(array,'array')) {
+    return "Wrong Data Type"
+  }
+
+let result = []
+  for(let i=0;i<array.length;i++) {
+    let intersection = false
```
Comment (Anasauce) [inline src/setIntersection.js:10]: again, no need to define a temp variable for ```intersection```

if the the ```if``` condition is true then just push to your array.  

If you want to only push the first instance that this occurs then I believe you can ```return``` inside the the if conditional inside your for loop and that breaks you out of the loop.  I know for certain that returning inside an if conditional always ends the function. I am 95% certain this also happens inside a for loop.

**Review by Anasauce — COMMENTED** [review]
Overall well-written tests. Some refactoring in your functions would make your code a little more concise. Try to use space between your operators and stick with the same stylistic spacing through out project for readability.
